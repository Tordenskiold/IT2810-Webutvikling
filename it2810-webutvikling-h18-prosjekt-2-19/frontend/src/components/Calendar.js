import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import DatePicker from 'react-datepicker';

import 'react-datepicker/dist/react-datepicker.css';
import CustomInput from './CustomInput';

export default class Calendar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      date: moment()
    };
  }

  formatedDate = (changedDate) => {
    const d = new Date(changedDate);
    this.setState({
      date: changedDate
    });

    // https://stackoverflow.com/a/30272803/597642
    return `${d.getFullYear()}-${`0${d.getMonth() + 1}`.slice(-2)}-${`0${d.getDate()}`.slice(-2)}`;
  };

  render() {
    const { onSelect } = this.props;
    const { date } = this.state;

    return (
      <DatePicker
        customInput={<CustomInput onClick={this.onSelect} value={date} />}
        selected={date}
        onChange={changedDate => onSelect(this.formatedDate(changedDate))}
        showYearDropdown
        dropdownMode="select"
      />
    );
  }
}

Calendar.propTypes = {
  onSelect: PropTypes.func.isRequired
};
