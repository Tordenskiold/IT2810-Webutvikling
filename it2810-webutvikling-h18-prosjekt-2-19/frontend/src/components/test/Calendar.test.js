import React from 'react';
import Calendar from '../Calendar';
import renderer from 'react-test-renderer';

//Jest test for Calendar.js
//Creates a snapshot test which tests the rendering
describe("Testing Calendar-component", () => {
  const component = renderer.create(<Calendar />);
  const tree = renderer.create(<Calendar />).getInstance();

  test("renders correctly", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});