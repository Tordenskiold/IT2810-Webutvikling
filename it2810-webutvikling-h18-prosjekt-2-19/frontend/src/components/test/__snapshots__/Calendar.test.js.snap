// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calendar-component renders correctly 1`] = `
<div
  className="react-datepicker-wrapper"
>
  <div
    className="react-datepicker__input-container"
  >
    <button
      className="example"
      onClick={[Function]}
      type="button"
    >
      11/16/2018
    </button>
  </div>
</div>
`;
